Comparing resource pools in tests is error prone

Shows up in `test_resource_pool_update_classes` occurs sometimes because it uses randomly generated inputs.

The problem is occuring because the tests compare list of resource classes and the lists contain the right/expected resource classes. It is just that the order is off.

```
@given(rp=rp_strat(), data=st.data())
      @settings(max_examples=5, suppress_health_check=[HealthCheck.function_scoped_fixture], deadline=None)
      @pytest.mark.asyncio
      async def test_resource_pool_update_classes(
          rp: models.ResourcePool, app_config: Config, data, admin_user: base_models.APIUser
      ) -> None:
          run_migrations_for_app("common")
          inserted_rp = None
          pool_repo = app_config.rp_repo
          try:
              inserted_rp = await create_rp(rp, pool_repo, api_user=admin_user)
              assert inserted_rp.id is not None
              old_classes = [asdict(cls) for cls in list(inserted_rp.classes)]
              new_classes_dicts = [{**cls, **data.draw(rc_update_reqs_dict)} for cls in old_classes]
              new_classes_models = [models.ResourceClass.from_dict(cls) for cls in new_classes_dicts]
              new_classes_models = sorted(new_classes_models, key=lambda x: (x.gpu, x.cpu, x.memory, x.max_storage, x.name))
      
              updated_rp = await pool_repo.update_resource_pool(
                  id=inserted_rp.id, classes=new_classes_dicts, api_user=admin_user
              )
      
              assert updated_rp.id == inserted_rp.id
              assert len(updated_rp.classes) == len(inserted_rp.classes)
  >           assert updated_rp.classes == new_classes_models
  E           AssertionError: assert [ResourceClas...lerations=[])] == [ResourceClas...346baa9455'])]
  E             
  E             At index 0 diff: ResourceClass(name='00000', cpu=2.0, memory=0, max_storage=1, gpu=0, id=883, default=False, default_storage=1, matching=None, node_affinities=[], tolerations=['e3e70682-c209-4cac-a29f-6fbed82c07cd', 'eb1167b3-67a9-4378-bc65-c1e582e2e662', 'f728b4fa-4248-4e3a-8a5d-2f346baa9455']) != ResourceClass(name='00000', cpu=1.0, memory=0, max_storage=1, gpu=0, id=884, default=True, default_storage=1, matching=None, node_affinities=[], tolerations=[])
  E             
  E             Full diff:
  E               [
  E             -     ResourceClass(
  E             -         name='00000',...
  E             
  E             ...Full output truncated (42 lines hidden), use '-vv' to show
```

--- a/components/renku_data_services/crc/models.py
+++ b/components/renku_data_services/crc/models.py
@@ -212,18 +212,17 @@ class ResourcePool:
         if not isinstance(other, ResourcePool):
             return False
 
-        if self.id != other.id:
-            return False
-        if self.name != other.name:
-            return False
-        if self.default != other.default or self.public != other.public:
-            return False
-        if self.idle_threshold != other.idle_threshold or self.hibernation_threshold != other.hibernation_threshold:
-            return False
-
-        this_classes = sorted(self.classes, key=lambda x: (x.default, x.cpu, x.memory, x.default_storage, x.name))
-        other_classes = sorted(other.classes, key=lambda x: (x.default, x.cpu, x.memory, x.default_storage, x.name))
-        return this_classes == other_classes
+        return (
+            self.id == other.id
+            and self.name == other.name
+            and self.default == other.default
+            and self.public == other.public
+            and self.idle_threshold == other.idle_threshold
+            and self.hibernation_threshold == other.hibernation_threshold
+            and self.quota == other.quota
+            and sorted(self.classes, key=lambda x: x.id or x.name)
+            == sorted(other.classes, key=lambda x: x.id or x.name)
+        )
 
     def set_quota(self, val: Quota) -> "ResourcePool":
         """Set the quota for a resource pool."""

--- a/test/components/renku_data_services/db/test_sqlalchemy_pool_repo.py
+++ b/test/components/renku_data_services/db/test_sqlalchemy_pool_repo.py
@@ -119,11 +119,16 @@ async def test_resource_pool_update_classes(
 
         assert updated_rp.id == inserted_rp.id
         assert len(updated_rp.classes) == len(inserted_rp.classes)
-        assert updated_rp.classes == new_classes_models
+        assert sorted(
+            updated_rp.classes,
+            key=lambda x: x.id or x.name,
+        ) == sorted(
+            new_classes_models,
+            key=lambda x: x.id or x.name,
+        )
         retrieved_rps = await pool_repo.get_resource_pools(id=inserted_rp.id, api_user=admin_user)
         assert len(retrieved_rps) == 1
         assert retrieved_rps[0] == updated_rp
-        assert retrieved_rps[0].classes == updated_rp.classes
     except (ValidationError, errors.ValidationError):
         pass
     finally:

